### PR TITLE
Closes #3: Add Psalm integration

### LINKED ISSUE #3: Psalm integration
### Feature Request

|    Q        |   A
|------------ | ------
| QA    | yes

#### Summary

As decided during the [Technical-Steering-Committee Meeting on August 3rd, 2020](https://github.com/laminas/technical-steering-committee/blob/main/meetings/minutes/2020-08-03-TSC-Minutes.md#vote-on-static-analysis-tooling), Laminas wants to implement [vimeo/psalm](https://github.com/vimeo/psalm) in all packages.

Implementing psalm is quite easy.

#### Required
- [ ] Create a `.psalm.xml.dist` in the project root
- [ ] Copy and paste the contents from this [psalm.xml.dist](https://github.com/mezzio/repo-template/blob/6124090610ef1c6d6b0c80a063cafacc3ec68f38/psalm.xml.dist)
- [ ] Run `$ composer require vimeo/psalm`
- [ ] Run `$ vendor/bin/psalm --set-baseline=psalm-baseline.xml`
- [ ] Add a composer script `static-analysis` with the command `psalm --shepherd --stats`
- [ ] Add a new line to `script:` in `.travis.yml`: `- if [[ $TEST_COVERAGE == 'true' ]]; then composer static-analysis ; fi`
- [ ] Remove phpstan from the project (`phpstan.neon.dist`, `.travis.yml` entry, `composer.json` `require-dev` and `scripts`)

##### Optional
- [ ] Fix as many psalm errors as possible.

### PR DESCRIPTION
**Pull request by alexraputa**
Closes #3

|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | yes
| RFC           | no
| QA            | yes

### CODE REVIEW

File: src/TemplatePath.php [hunk 1]
```diff
@@ -6,11 +6,9 @@
 
 class TemplatePath
 {
-    /** @var string */
-    protected $path;
+    protected string $path;
```
Comment (samsonasik) [inline src/TemplatePath.php]: Php 7.3 is not typed property yet, also update to typed property should only for private modifier as can cause bc break for protected and public even on php 7.4 if the version change is minor, also, this is out of the PR scope.

Comment (alexraputa) [reply src/TemplatePath.php]: Done.

**Comment by boesing** [comment]
@alexraputa Thanks, could you please rebase against the 2.3.x branch? I've released 2.2.x with PHP 8.1 support.

Thanks!

**Comment by alexraputa** [comment]
@boesing, Done 🚀

**Comment by alexraputa** [comment]
@boesing friendly ping ;)

**Comment by froschdesign** [comment]
@alexraputa 
It is not necessary to add a comment like this, this request is included in the relevant lists here on GitHub and will be handled when time is available.
Thank you for your understanding!

**Comment by alexraputa** [comment]
@froschdesign
Oh, I'm sorry. I don't know your workflow.

**Review by Ocramius — APPROVED** [review]
LGTM :+1:

**Comment by Ocramius** [comment]
Thanks @alexraputa!

**Comment by froschdesign** [comment]
@alexraputa 
It's a typical open source project, most of us have full-time jobs and it's not Laminas or Mezzio.
We use [the projects here on GitHub](https://github.com/orgs/mezzio/projects) for example, so your contribution will not be forgotten.

Sometimes you have to wait and sometimes it happens very quickly: thanks @Ocramius 👍

**Pull request merged**